src/propterm.js

```diff
@@ -21,593 +21,667 @@
  */
 
 
-
-/**
- *
- * @type {number}
- */
-var cursorX = 0;
-
-
-/**
- *
- * @type {number}
- */
-var cursorY = 0;
-
-
-/**
- *
- * @type {number}
- */
-var cursorGotoX = 0;
-
-
-/**
- *
- * @type {number}
- */
-var cursorGotoY = 0;
-
-
 /**
- *
- * @type {number}
- * 0 = normal, 1 = set X, 2 = set Y, 3 = set X then to 2 to set Y
+ * @description A lightweight, generic in-browser terminal with cursor positioning, clearing, and text-wrapping support.
+ * @author Matthew Matz matt.m.matz@gmail.com
+ * @function PropTerm
+ * @param {object} terminalElement HTML element to populate the terminal into.
+ * @param {object} outputCallback function to call and send characters typed into the terminal.
+ * @param {object} options options for configuring the terminal.
```

Comment:
The 'options' object is a bit of a mystery. Is it only used with the PropTerm class? If so, the declaration and implementation should be moved into this file from wherever is exists now. (I can't seem to locate it.).

Reply:
When editor.js creates a new PropTerm (Line 398 - editor.js), we aren't using any options outside of the defaults.  We were at one time, because we were supporting older versions of the client that did echoing differently.  I baked the options in because I feel this thing could be a good standalone piece of software - yes, they are settings specifically for the PropTerm.

That being said - they could be stripped out of our implementation of it, since we are only using defaults.

I tried my best to document each option available and how they are used in the JSdoc at the top of the class.  That's why each option within the class is listed in the JSdoc params - I wasn't sure how else to document them.